### ISSUE TITLE
build failure in quasi_macros

### ISSUE THREAD

**Issue by sourcefrog** [Byron/google-apis-rs#131]
I'm trying to build google-apis-rs 0bd7f2004843b4e9dcd8af366e7ffc6632fb9e41:

It doesn't build on the stable or beta toolchain because of a feature gate in serde_codegen

```
   Compiling serde_codegen v0.6.1
/Users/mbp/.multirust/toolchains/beta/cargo/registry/src/github.com-0a35038f75765ae4/quasi_macros-0.3.5/src/lib.rs:11:1: 11:63 error: #[feature] may not be used on the beta release channel
/Users/mbp/.multirust/toolchains/beta/cargo/registry/src/github.com-0a35038f75765ae4/quasi_macros-0.3.5/src/lib.rs:11 #![feature(plugin_registrar, unboxed_closures, rustc_private)]
                                                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I don't know if you intend to support stable?

On nightly, I see a different failure:

```
/Users/mbp/.multirust/toolchains/nightly/cargo/registry/src/github.com-0a35038f75765ae4/quasi_macros-0.3.5/src/lib.rs:19:29: 19:32 error: mismatched types:
 expected `&mut syntex::Registry`,
    found `&mut rustc::plugin::registry::Registry<'_>`
(expected struct `syntex::Registry`,
    found struct `rustc::plugin::registry::Registry`) [E0308]
/Users/mbp/.multirust/toolchains/nightly/cargo/registry/src/github.com-0a35038f75765ae4/quasi_macros-0.3.5/src/lib.rs:19     quasi_codegen::register(reg);
                                                                                                                                                     ^~~
/Users/mbp/.multirust/toolchains/nightly/cargo/registry/src/github.com-0a35038f75765ae4/quasi_macros-0.3.5/src/lib.rs:19:29: 19:32 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
```

I reported this as https://github.com/serde-rs/quasi/issues/15 but it doesn't reproduce in the quasi_macros tree built by itself.

**Comment by Byron** [comment]
This project is supposed to build on both stable and nightly. And even though nightly can break any time primarily due to serde, stable should be ... stable :).

However, it works perfectly fine for me on stable:

```bash
$ make youtube3-cli-cargo ARGS=check
```

```bash
➜  google-apis-rs git:(master) rustc --version --verbose
rustc 1.3.0 (9a92aaf19 2015-09-15)
binary: rustc
commit-hash: 9a92aaf19a64603b02b4130fe52958cc12488900
commit-date: 2015-09-15
host: x86_64-apple-darwin
release: 1.3.0
```

Maybe you have a local override of some sort ?
Otherwise, could you give me a hint on how to reproduce the issue ?

**Comment by erickt** [comment]
I fixed this in #132. The problem is that there are conflicting versions of `serde_codegen` being imported. `yup-oauth2` is pulling in the version built with `syntex`, but there's also a dependency on `serde_macros` which is built with `libsyntax`.
